interface_adapter/loggedin/LoggedInViewModel.java

```diff
@@ -35,4 +59,15 @@ public void addPropertyChangedListener(PropertyChangeListener newListener) {
     public LoggedInState getWindowState() {
         return windowState;
     }
+
```

Comment:
It might make sense to move this to another class called ImageScaler to better follow SOLID principles (SRP). You could also change the method to include optional width and height parameters.